sunpy/map/mapbase.py

```diff
@@ -631,14 +631,16 @@ def observer_coordinate(self):
         """
         The Heliographic Stonyhurst Coordinate of the observer.
         """
+        missing_meta = {}
         for keys, kwargs in self._supported_observer_coordinates:
-            if all([k in self.meta for k in keys]):
+            meta_list = [k in self.meta for k in keys]
+            if all(meta_list):
                 return SkyCoord(obstime=self.date, **kwargs).heliographic_stonyhurst
+            elif any(meta_list) and not set(keys).isdisjoint(self.meta.keys()):
+                missing_meta[kwargs['frame']] = set(keys).difference(self.meta.keys())
 
-        all_keys = [str(e[0]) for e in self._supported_observer_coordinates]
-        all_keys = '\n'.join(all_keys)
-        warning_message = ("Missing metadata for observer: assuming Earth-based observer."
-                           "The following sets of keys were checked:\n" + all_keys)
+        warning_message = "".join([f"For frame '{frame}' the following metadata is missing :{','.join(keys)}\n" for frame, keys in missing_meta.items()])
```

Comment:
I think it would look better with the space the other side of the colon:

```suggestion
        warning_message = "".join([f"For frame '{frame}' the following metadata is missing: {','.join(keys)}\n" for frame, keys in missing_meta.items()])
```

Reply:
Done